pkg/image/registry/image/strategy.go

```diff
@@ -127,9 +133,15 @@ func (s imageStrategy) PrepareForUpdate(ctx apirequest.Context, obj, old runtime
 		newImage.DockerImageManifest = ""
 		newImage.DockerImageConfig = ""
 	}
+
+	// bug in image update logic allowed to set arbitrary annotations that would otherwise be rejected by
+	// validation; resolves rhbz#1557607
+	for i := range newImage.Signatures {
+		newImage.Signatures[i].Annotations = map[string]string{}
```

Comment:
I would only drop the old controller annotation.

Reply:
yeah i guess that makes sense.  for any other annotation they'll run into the validation rule.

Reply:
Fixed.